crowbar_framework/app/models/node_object.rb

```diff
@@ -724,7 +724,11 @@ def rebuild_run_list
     crowbar["run_list_map"] = {} if crowbar["run_list_map"].nil?
 
     # Cull by state
-    map = crowbar["run_list_map"].select { |k,v| v["states"].include?("all") or v["states"].include?(self.crowbar["state"]) }
+    map = crowbar["run_list_map"].select do |k, v|
+      v["states"].include?(state) ||
```

Comment:
As mentioned in https://github.com/crowbar/barclamp-crowbar/pull/1362, I think using `state` is wrong.